src/main/java/org/truffleruby/core/regexp/MatchDataNodes.java

```diff
@@ -282,14 +285,62 @@ protected boolean inBounds(DynamicObject matchData, int index) {
     }
 
 
+    @NodeChildren({
+            @NodeChild(type = RubyNode.class, value = "matchData")
+    })
```

Comment:
No need for `NodeChildren` anymore in Java 8.

Reply:
But normally with just use CoreMethodArrayArgumentsNode unless there is a need to name the children for a `@CreateCast`.